onionshare/mode_settings.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+"""
+OnionShare | https://onionshare.org/
+
+Copyright (C) 2014-2018 Micah Lee <micah@micahflee.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import os
+import pwd
+import json
+
+
+class ModeSettings:
+    """
+    This stores the settings for a single instance of an OnionShare mode. In CLI there
+    is only one TabSettings, and in the GUI there is a separate TabSettings for each tab
```

Comment:
'TabSettings' or 'ModeSettings' ?

Reply:
Good catch. I called this `TabSettings` at first, but renamed it to `ModeSettings` because it made more sense, especially since CLI doens't have tabs but still uses it.